jsonrpc/filter_manager.go

```diff
@@ -818,14 +809,23 @@ func (t *timeHeapImpl) Pop() interface{} {
 // of the stream at any point
 type blockStream struct {
 	lock sync.Mutex
```

Comment:
Do we still need lock here, since we have introduced `atomic`?

Reply:
Let me try, the test for concurrency is better now.

Reply:
So, the lock is there to avoid concurrent `push`. In theory, there cannot be any because of how `push` is being used, but I wanted to be sure.

Reply:
> So, the lock is there to avoid concurrent push.

What do you think about keeping lock used only in `push` and remove it from `Head`, since `Head` completely relies on atomic (just loading a value)?

Reply:
You got it! I also removed the lock in Head.